src/transformers/models/fuyu/image_processing_fuyu.py

```diff
@@ -51,76 +58,144 @@ class FuyuImageProcessor(BaseImageProcessor):
         """Calculate number of patches required to encode an image."""
-        if img_h % patch_dim_h != 0:
-            raise ValueError(f"{img_h=} must be divisible by {patch_dim_h=}")
-        if img_w % patch_dim_w != 0:
-            raise ValueError(f"{img_w=} must be divisible by {patch_dim_w=}")
+        if image_height % self.patch_size["height"] != 0:
+            raise ValueError(f'{image_height=} must be divisible by {self.patch_size["height"]}')
+        if image_width % self.patch_size["width"] != 0:
+            raise ValueError(f'{image_width=} must be divisible by {self.patch_size["width"]}')
 
-        num_patches_per_dim_h = img_h // patch_dim_h
-        num_patches_per_dim_w = img_w // patch_dim_w
+        num_patches_per_dim_h = image_height // self.patch_size["height"]
+        num_patches_per_dim_w = image_width // self.patch_size["width"]
         num_patches = num_patches_per_dim_h * num_patches_per_dim_w
 
         return num_patches
 
-    def patchify_image(self, image: "torch.Tensor", patch_dim_h: int, patch_dim_w: int) -> "torch.Tensor":
+    def patchify_image(self, image: "torch.Tensor") -> "torch.Tensor":
         """
         Convert an image into a tensor of patches.
 
         Args:
             image: Image to convert. Shape: [batch, channels, height, width]
-            patch_dim_h: Height of each patch.
-            patch_dim_w: Width of each patch.
         """
         requires_backends(self, ["torch"])
 
         # TODO refer to https://github.com/ArthurZucker/transformers/blob/0f0a3fe5ca5697ee58faeb5b53f049af720b5e98/src/transformers/models/vit_mae/modeling_vit_mae.py#L871
         # torch implementation is faster but does not handle non-squares
 
-        batch_size, channels, height, width = image.shape
-        unfolded_along_height = image.unfold(2, patch_dim_h, patch_dim_h)
-        patches = unfolded_along_height.unfold(3, patch_dim_w, patch_dim_w)
+        batch_size, channels, _, _ = image.shape
+        unfolded_along_height = image.unfold(2, self.patch_size['height'], self.patch_size['height'])
+        patches = unfolded_along_height.unfold(3, self.patch_size["width"], self.patch_size["width"])
 
-        patches_reshaped = patches.contiguous().view(batch_size, channels, -1, patch_dim_h, patch_dim_w)
+        patches_reshaped = patches.contiguous().view(
+            batch_size, channels, -1, self.patch_size['height'], self.patch_size["width"])
 
         patches_final = patches_reshaped.permute(0, 2, 3, 4, 1).reshape(
-            batch_size, -1, channels * patch_dim_h * patch_dim_w
+            batch_size, -1, channels * self.patch_size['height'] * self.patch_size["width"]
         )
 
         return patches_final
 
-    def process_images_for_model_input(
+    def preprocess(self, images):
+        """Utility function to preprocess the images and extract necessary information about original formats."""
+        batch_images = []
```

Comment:
Should this be a tensor instead of a list? I'm asking because it makes it harder to change the `device` or `dtype` before generation.